torchvision/models/video/mvit.py

```diff
@@ -177,24 +261,53 @@ def __init__(
                 norm_layer(self.head_dim),
             )
 
+        self.rel_pos_h: Optional[nn.Parameter] = None
+        self.rel_pos_w: Optional[nn.Parameter] = None
+        self.rel_pos_t: Optional[nn.Parameter] = None
+        if rel_pos:
+            if input_size[1] != input_size[2]:
+                raise ValueError("Relative Positional Embeddings require square input shape.")
+            size = input_size[1]
```

Comment:
@haooooooqi @lyttonhao @feichtenhofer I would love to get your input as authors of the paper on how one could remove this limitation from the implementation. Ideally we would like to support any arbitrary rectangle. Do you have an advice here? Perhaps `size = max(input_size[1:])` could work?

Reply:
Yeah,  using `max(input_size[1:])` could work or calculating the size for h and w separately should also work.

Reply:
Thanks a lot for the prompt reply. I'll make the necessary change.